### PR TITLE
Add infrastructure for integrating BTOR2 interpreter into Calyx

### PR DESCRIPTION
**Pull request by obhalerao**
Add the BTOR2 interpreter along with infrastructure necessary for integrating the interpreter with Calyx. I also created a dummy BTOR2Add primitive that reads a hardcoded program from disk and tested it against Cider 2 to ensure correctness of the integration.

### CODE REVIEW

**Comment by rachitnigam** [comment]
This has a lot of generated files (`.btor2`). Is that intentional? CC @EclecticGriffin

**Comment by obhalerao** [comment]
> This has a lot of generated files (`.btor2`). Is that intentional? CC @EclecticGriffin

Ah; those files were tests from the old repo I imported the interpreter from. I can remove/refactor those.

File: tools/btor2/btor2i/src/values.rs
Comment (EclecticGriffin) [inline tools/btor2/btor2i/src/values.rs:1]: Don't commit this file since it isn't included in the module tree (and won't compile if it is included)

**Review by EclecticGriffin — COMMENTED** [review]
Looks good for the most part. Left a few minor nits. I didn't look through much of the BTOR2i code but I can do that later if desired.

**Pull request merged**